internal/cel/library/crypto.go

```diff
@@ -0,0 +1,137 @@
+package library
+
+import (
+	"reflect"
+
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/common/types"
+	"github.com/google/cel-go/common/types/ref"
+	"github.com/google/cel-go/common/types/traits"
+	"github.com/kubewarden/policy-sdk-go/pkg/capabilities/crypto"
+)
+
+// Crypto returns a cel.EnvOption to configure namespaced crypto host-callback
+// Kubewarden functions.
+//
+// # Crypto.VerifyCert
+//
+// This CEL function accepts a certificate, a certificate chain, and an
+// expiration date.
+// It returns a bool on whether the provided CertificateVerificationRequest
+// (containing a cert to be verified, a cert chain, and an expiration date)
+// passes certificate verification.
+//
+// Accepts 3 arguments:
+//   - string,  of PEM-encoded certificate to verify.
+//   - list of strings, of PEM-encoded certs, ordered by trust usage
+//     (intermediates first, root last). If empty, certificate is assumed trusted.
+//   - string in RFC 3339 time format, to check expiration against.
+//     If empty, certificate is assumed never expired.
+//
+// Returns a map(<string>) with 2 fields:
+//   - "Trusted": <bool> informing if certificate passed verification or not
+//   - "Reason": <string> with reason, in case "Trusted" is false
+//
+// Usage in CEL:
+//
+//	crypto.verifyCert(<string>, list(<string>), <string>) -> map(<string>, value)
+//
+// Example:
+//
+//	  kw.crypto.verifyCert(
+//				 '---BEGIN CERTIFICATE---foo---END CERTIFICATE---',
+//		    [
+//		      '---BEGIN CERTIFICATE---bar---END CERTIFICATE---'
+//		    ],
+//		    '2030-08-15T16:23:42+00:00'
+//		 )"
+func Crypto() cel.EnvOption {
+	return cel.Lib(cryptoLib{})
+}
+
+type cryptoLib struct{}
+
+// LibraryName implements the SingletonLibrary interface method.
+func (cryptoLib) LibraryName() string {
+	return "kw.crypto"
+}
+
+// CompileOptions implements the Library interface method.
+func (cryptoLib) CompileOptions() []cel.EnvOption {
+	return []cel.EnvOption{
+		// group every binding under a container to simplify usage
+		cel.Container("crypto"),
+
```

Comment:
we are not using the request/response struct native types.
also, what about DER certificates?

Reply:
> also, what about DER certificates?

Since Der is a binary data format that would not be fit for a string, it seemed a good idea to not provide it. 

> we are not using the request/response struct native types.

If we don't need a type `Certificate{}` with an `Encoding` field for Der, we don't need the request types.

Also, we couldn't use the Request type as-is, as the struct type `capabilities/crypto/Certificate{}` expects a []rune (and by extension the struct type `CertificateVerificationRequest{}`). CEL only knows about strings, bytes, and []int, and provides stringToBytes (not fit) and stringToInt64 (should be int8). We would need intermediate helper types.

For the Response type, it could be done, but at that point I preferred to return a map than a CertificateVerificationResponse{} that needs to be documented. It may also seem simpler to CEL users.

Reply:
@fabriziosestito it's my fault, I told Victor this could have been dropped because it would not be possible to write binary data into the definition of a policy.

However, @fabriziosestito reminded me CEL policies can also fetch data from external sources, like a ConfigMap. Do you think this could be a way to make DER data appear into a policy? I suspect you would still have to take the DER data, encode it using base64 and add that into a Secrets/ConfigMap. The CEL policy would have to work in reverse order: base64 decode (is there a way to do that) -> DER -> our function.

Is that something that can realistically happen? Should we just focus on the most common use case right now (PEM) and expand to support DER later on?
I'm a bit conflicted, because if we were to add DER support later on, we would have to break our CEL API; which would be super painful...

Reply:
We could provide 2 different CEL functions, one for Pem and one for Der. I could rename the current function.

Reply:
Opened https://github.com/kubewarden/cel-policy/issues/23.

I'm conflicted if to subtly change the current `kw.crypto.verifyCert()` signature to make it more easier to add DER certs later on.